wg-manifests/charter.md

```diff
@@ -0,0 +1,100 @@
+# WG Manifests Charter
+
+This charter adheres to the conventions, roles and organization management
+outlined in [wg-governance].
+
+## Scope
+
+- Provide a catalog (centralized repository) of Kubeflow application manifests.
+- Provide a catalog of third-party apps for common services.
+- Provide documentation to help users install a set, or all of the included
+  apps.
+
+### In scope
+
+#### Code, Binaries and Services
+
+- Maintain a set of manifests that will allow users to install Kubeflow apps and
+  common services easily on Kubernetes, either on the cloud or on-prem, without
+  depending on external cloud services or closed source solutions. These
+  manifests will be developed with certain principles in mind, which are
+  outlined in the last bullet. Those include:
+    1. A common set of manifests for the current five (5) official Kubeflow
+       applications:
+        - Training Operators
+        - Kubeflow Pipelines (KFP)
+        - Notebooks
+        - KFServing
+        - Katib
+    1. Manifests for a set of specific common services:
+       - Istio
+       - KNative
+       - Dex
+       - Cert-Manager
+    1. Manifests for common, currently unmaintained, apps:
+        - Central Dashboard
+        - Profile Controller
+        - PodDefaults Controller
+- Maintain documentation and instructions for installing a set or all of the
+  apps. Documentation will not depend on any opinionated deployment tool. It
+  will only depend on `kustomize` and `kubectl`. This way, users and platform
+  owners can easily build on top and use their favorite opinionated deployment
+  tool (e.g., Argo, kapp, kpt, kfctl, FluxCD, etc.) with Kubeflow.
+- The maintained manifests will follow certain principles:
+    - The flow of work / separation of responsibilities will be the following:
+        1. Application owners publish manifests in their repos.
+        1. WG copies and tracks upstream manifests in the manifests repo. They
+           form a base `kustomization`.
+        1. All kubeflow-specific changes (e.g., change the namespace) are done
```

Comment:
I don't understand this. What are "kubeflow-specific" changes? If an application should be installed in the "kubeflow" namespace shouldn't that be part of the application manifest and maintained in the upstream repo.

Reply:
@jlewi what I mean is that some applications are installed in a slightly different ways sometimes.
For example, KFServing in manifests used the kubeflow namespace, while the standalone in the upstream repo used the kfserving-system namespace.

Now, this particular example no longer applies, as KFServing now also maintains a kubeflow overlay which deploys in the `kubeflow` namespace, but the fact remains.

So I will use another example. The user identity header expected by an application is not the same across the board.
- Jupyter Web App expects `kubeflow-userid` (https://github.com/kubeflow/kubeflow/blob/6eb007d0b7b37ece0eb9a072f12274aab0f8fe09/components/crud-web-apps/common/backend/kubeflow/kubeflow/crud_backend/authn.py#L12-L13)
- Pipelines expects `x-goog-authenticated-user-email` (https://github.com/kubeflow/pipelines/blob/5435e8724fe1b2c6175f219217c82dee33227b4b/frontend/server/configs.ts#L99)

So for Pipelines and JWA to work together, we have to tweak one of the two slightly.
This change can happen in an overlay, maintained by wg-manifests, so that the user can install pipelines, jwa, or the two of them together.

Ideally, everything will be kind of uniform on the app side, so we won't have to maintain these, but unfortunately this is not always the case. We want to be able to do these minor fixes if needed.

Reply:
Thanks @yanniszark Per [comment](https://github.com/kubeflow/community/pull/402#issuecomment-682704600) I would expect this to work as follows.

IIUC the header will be different depending on the distribution. With Google IAP the header gets set to "x-goog-authenticated-user-email" with Dex or Cognito I presume its a different a header.

Its up to each application to make the header configurable; e.g. provide an environment variable that allows the header to be set.

It would then be up to the distribution to set the environment variable correctly for all applications.

So lets suppose you wanted to define an overlay to make the environment variable configurable; Per [comment](https://github.com/kubeflow/community/pull/402#issuecomment-682704600) I  think we should pursue one of the following

1. We push that overlay upstream into the app's repo upstream of manifests
1. We push the overlay downstream into the distribution's repo

How does that compare to what you are thinking?

Reply:
I agree that ideally we would want all these overlays to be patched on the upstream apps, but we need to have a way to be able to push them to the manifests repo, even in another directory, so we can actually test. And I agree that we should then try to then push upstream or downstream to have the ideal scenario, where this directory is empty.